src/sources/soundsourceffmpeg.cpp

```diff
@@ -22,6 +25,70 @@ void initFFmpegLib() {
     avcodec_register_all();
 }
 
+// More than 2 channels are currently not supported
+const SINT kMaxChannelCount = 2;
+
+inline
+AVMediaType getMediaType(AVStream* pStream) {
+    return pStream->codec->codec_type;
+}
+
+AVStream* findNextAudioStream(AVFormatContext* pFormatCtx, AVStream* pStream = nullptr) {
+    DEBUG_ASSERT(pFormatCtx != nullptr);
+    DEBUG_ASSERT((pStream == nullptr) || (static_cast<unsigned int>(pStream->index) < pFormatCtx->nb_streams));
+    DEBUG_ASSERT((pStream == nullptr) || (pStream == pFormatCtx->streams[pStream->index]));
+    unsigned int iNextStream;
+    if (pStream == nullptr) {
+        // Start search at the first stream
+        iNextStream = 0;
+    } else {
+        // Restart search at the following stream
+        iNextStream = pStream->index + 1;
+    }
+    while (iNextStream < pFormatCtx->nb_streams) {
+        AVStream* pNextStream = pFormatCtx->streams[iNextStream];
+        if (getMediaType(pNextStream) == AVMEDIA_TYPE_AUDIO) {
+            return pNextStream;
+        } else {
+            // Continue search at the next stream
+            ++iNextStream;
+        }
+    }
+    return nullptr;
+}
+
+inline
+AVStream* findFirstAudioStream(AVFormatContext* pFormatCtx) {
+    return findNextAudioStream(pFormatCtx);
+}
+
+inline
+AVCodec* findDecoder(AVStream* pStream) {
+    return avcodec_find_decoder(pStream->codec->codec_id);
+}
+
+inline
+SINT getChannelCount(AVStream* pStream) {
+    return pStream->codec->channels;
+}
+
+inline
+SINT getSamplingRate(AVStream* pStream) {
+    return pStream->codec->sample_rate;
+}
+
+inline
+SINT getFrameCount(AVStream* pStream) {
+    // NOTE(uklotzde): Use 64-bit integer calculation to minimize rounding errors
+    return (static_cast<uint64_t>(pStream->time_base.num) * static_cast<uint64_t>(pStream->duration) * static_cast<uint64_t>(getSamplingRate(pStream))) /
+            static_cast<uint64_t>(pStream->time_base.den);
```

Comment:
pStream->duration is already int64_t, a uint64_t ignores the sign. 
no need to cast getSamplingRate(pStream) int64_t * int is int64_t
It looks like, you need no cast at all to have maximum precision.